cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/OffsetManagementUtil.java

```diff
@@ -0,0 +1,91 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.monitor;
+
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.internal.app.store.AppMetadataStore;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Util to manage offsets for runtime(s).
+ */
+public class OffsetManagementUtil {
+  private static final String SEPERATOR = ":r:";
+
+  /**
+   * Initialize offsets for given runtime.
+   */
+  public static void initializeOffsets(CConfiguration cConf, DatasetFramework dsFramework, DatasetContext context,
+                                       Map<String, MonitorConsumeRequest> initialOffsets, String programRunId,
+                                       Set<String> topicConfigs) throws Exception {
+    int limit = cConf.getInt(Constants.RuntimeMonitor.BATCH_LIMIT);
+
+    for (String topicConfig : topicConfigs) {
+      String messageId = getAppMetadataStore(cConf, dsFramework, context).retrieveSubscriberState(
+        getRowKey(programRunId, topicConfig), null);
+
+      initialOffsets.put(topicConfig, new MonitorConsumeRequest(messageId, limit));
+    }
+  }
+
+  /**
+   * Persists offsets for given runtime for each topic {@link RuntimeMonitor} is monitoring
+   */
+  public static void persistOffsets(CConfiguration cConf, DatasetFramework dsFramework, DatasetContext context,
+                                    String programRunId, String topicConfig, String messageId) throws Exception {
+    getAppMetadataStore(cConf, dsFramework, context).persistSubscriberState(getRowKey(programRunId, topicConfig),
+                                                                            null, messageId);
+  }
+
+  /**
+   * Cleans up offsets persisted for given runtime for each topic {@link RuntimeMonitor} is monitoring
+   */
+  public static void cleanupOffsets(CConfiguration cConf, DatasetFramework dsFramework, DatasetContext context,
+                                    String programRunId) throws Exception {
+    getAppMetadataStore(cConf, dsFramework, context).deleteSubscriberState(getRowKeyPrefix(programRunId), null);
+  }
+
+  /**
+   * Creates App metadata store instance.
+   */
+  private static AppMetadataStore getAppMetadataStore(CConfiguration cConf, DatasetFramework dsFramework,
```

Comment:
Avoid one line method like this that do nothing other than just calling another method with exactly the same arguments.